Source/Modules/scilab.cxx

```diff
@@ -938,33 +872,51 @@ class SCILAB:public Language {
     for (int i = 0; i < Len(sourceFileList); i++) {
       String *sourceFile = Getitem(sourceFileList, i);
       if (i == 0) {
-	Printf(builderCode, "files = \"%s\";\n", sourceFile);
+	      Printf(builderCode, "files = \"%s\";\n", sourceFile);
       } else {
-	Printf(builderCode, "files($ + 1) = \"%s\";\n", sourceFile);
+	      Printf(builderCode, "files($ + 1) = \"%s\";\n", sourceFile);
       }
     }
 
-    Printf(builderCode, "table = [");
+    Printf(builderCode5, "table = [");
+    Printf(builderCode6, "table = [");
   }
 
   /* -----------------------------------------------------------------------
    * addFunctionInBuilderCode()
    * Add a function wrapper in the function table of generated builder script
    * ----------------------------------------------------------------------- */
 
-  void addFunctionInScriptTable(const_String_or_char_ptr scilabFunctionName, const_String_or_char_ptr wrapperFunctionName, String *scriptCode) {
+  void addFunctionInScriptTable(const_String_or_char_ptr scilabFunctionName, const_String_or_char_ptr scilabSmallFunctionName, const_String_or_char_ptr wrapperFunctionName, String *scriptCode5, String *scriptCode6) {
     if (++builderFunctionCount % 10 == 0) {
-      Printf(scriptCode, "];\ntable = [table;");
+      Printf(scriptCode5, "];\ntable = [table;");
+      Printf(scriptCode6, "];\ntable = [table;");
     }
-    Printf(scriptCode, "\"%s\",\"%s\";", scilabFunctionName, wrapperFunctionName);
+    Printf(scriptCode5, "\"%s\",\"%s\";", scilabSmallFunctionName, wrapperFunctionName);
+    Printf(scriptCode6, "\"%s\",\"%s\";", scilabFunctionName, wrapperFunctionName);
   }
 
   /* -----------------------------------------------------------------------
    * saveBuilderFile()
    * ----------------------------------------------------------------------- */
 
   void saveBuilderFile(String *gatewayName) {
-    Printf(builderCode, "];\n");
+    Printf(builderCode5, "];\n");
+    Printf(builderCode6, "];\n");
+
+    Printf(builderCode, "ver = getversion('scilab');\n");
+    Printf(builderCode, "if ver(1) < 6 then\n");
+    Printf(builderCode, "  // version is less or equal to 5.5.2\n");
+    Printf(builderCode, "  \n");
+    Append(builderCode, builderCode5);
+    Printf(builderCode, "  \n");
+    Printf(builderCode, "else\n");
+    Printf(builderCode, "  // version is 6.0.0 or more\n");
+    Printf(builderCode, "  \n");
+    Append(builderCode, builderCode6);
+    Printf(builderCode, "  \n");
+    Printf(builderCode, "end\n");
```

Comment:
If I follow, `builderCode5` is essentially the same as `builderCode6` except that identifiers are truncated (or slightly a slightly more complex adjustment for some cases) to be at most 24 characters.

Does that mean that if there aren't any long identifiers in an API that `builderCode5` will be identical to `builderCode6`?

If so, it seems worthwhile to have a special-case for that as it should give a smaller wrapper in many common cases - something like:

```
    if (Equal(builderCode5, builderCode6)) {
      Append(builderCode, builderCode6);
    } else {
      // The code above
    }
```

And probably similarly for `loaderScript5` vs `loaderScript6`.

Reply:
Good idea ! done in the latest push